baseapp/baseapp.go

```diff
@@ -66,7 +66,11 @@ type BaseApp struct {
 	deliverState *state          // for DeliverTx
 	voteInfos    []abci.VoteInfo // absent validators from begin block
 
-	// minimum fees for spam prevention
+	// consensus params
+	// TODO move this in the future to baseapp param store on main store.
+	consensusParams *abci.ConsensusParams
```

Comment:
So now we have this part of the BaseApp. But we also have `WithConsensusParams` on a context. This seems a bit confusing, no? Do we want both? Should we just remove `ConsensusParams` from the context?

Reply:
This is not the consensusParams, just a pointer to retrieve these params for the context in BeginBlock. I think it makes sense to leave as is and stay consistent with the context design pattern. - We'll also be removing this term from bassapp with a future refactor (https://github.com/cosmos/cosmos-sdk/issues/2882)

Reply:
Idk...it's just confusing to me tbh.

Reply:
Also, can we allow the author of the conversation to resolve? i.e. if there are things that still need further discussion.

Reply:
I concur with bez's point, having a pointer here feels odd, and the redundancy is confusing. If its still confusing you can unresolve as well (idk how we want to handle who clicks the resolve button initially)